openadapt/utils.py

```diff
@@ -412,17 +415,63 @@ def evenly_spaced(arr: list, N: list) -> list:
     return [val for idx, val in enumerate(arr) if idx in idxs]
 
 
+
+
+def get_current_monitor(monitors):
+    """Determine the monitor where the cursor is currently located.
+
+    Args:
+        monitors (list): List of monitor dictionaries.
+
+    Returns:
+        dict: The monitor dictionary where the cursor is located.
+    """
+    cursor_x, cursor_y = pyautogui.position()
+    
+    for monitor in monitors:
+        if monitor['left'] <= cursor_x < monitor['left'] + monitor['width'] and \
+           monitor['top'] <= cursor_y < monitor['top'] + monitor['height']:
+            return monitor
+    
+    # If not found, default to the first monitor
+    return monitors[1]
+
 def take_screenshot() -> Image.Image:
-    """Take a screenshot.
+    """Take a screenshot of the current monitor or all monitors.
 
     Returns:
-        PIL.Image: The screenshot image.
-    """
-    # monitor 0 is all in one
-    monitor = SCT.monitors[0]
-    sct_img = SCT.grab(monitor)
-    image = Image.frombytes("RGB", sct_img.size, sct_img.bgra, "raw", "BGRX")
-    return image
+        PIL.Image.Image: The screenshot image.
+    """
+    with mss.mss() as sct:
```

Comment:
Is it possible to re-use the global `SCT` here? Creating a new instance impacts performance; see https://python-mss.readthedocs.io/usage.html#intensive-use